core/src/Kokkos_Macros.hpp

```diff
@@ -603,8 +603,9 @@ static constexpr bool kokkos_omp_on_host() { return false; }
 // Guard intel compiler version 19 and older
 // intel error #2651: attribute does not apply to any entity
 // using <deprecated_type> KOKKOS_DEPRECATED = ...
-#if defined(KOKKOS_ENABLE_DEPRECATION_WARNINGS) && !defined(__NVCC__) && \
-    (!defined(KOKKOS_COMPILER_INTEL) || KOKKOS_COMPILER_INTEL >= 2021)
+#if defined(KOKKOS_ENABLE_DEPRECATION_WARNINGS) && !defined(__NVCC__) &&  \
+    (!defined(KOKKOS_COMPILER_INTEL) || KOKKOS_COMPILER_INTEL >= 2021) && \
+    !defined(KOKKOS_ENABLE_OPENACC)
```

Comment:
I find it interesting that it works in the NVHPC build and not in the OPENACC one.
They use nvc++ version 22.7 and 22.3 respectively.
Let's check if it works with more recent version of the compiler and bump the version if that happens to do the trick
@seyonglee @pedrovalerolara

Reply:
Nevermind I see the same issue with 22.9
We need to report this.

Reply:
#5604 fixes the issue